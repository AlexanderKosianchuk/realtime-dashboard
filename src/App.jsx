import { hot } from 'react-hot-loader/root';
import React, { PureComponent } from 'react';
import { Router, Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet-async';
import { ThemeProvider } from 'styled-components';

import history from 'modules/history';
import theme from 'modules/theme';

import Home from 'routes/Home';
import NotFound from 'routes/NotFound';

import GlobalStyles from 'controls/GlobalStyles';
import SystemAlerts from 'features/systemAlerts';

export class App extends PureComponent {
  render() {
    return (
      <Router history={history}>
        <ThemeProvider theme={theme}>
          <Helmet
            defer={false}
            htmlAttributes={{ lang: 'en-us' }}
            encodeSpecialCharacters={true}
            defaultTitle={'title'}
            titleTemplate={`%s | title`}
            titleAttributes={{ itemprop: 'name', lang: 'en-us' }}
          />
          <Switch>
            <Route path="/" exact component={Home} />
            <Route component={NotFound} />
          </Switch>
          <SystemAlerts />
          <GlobalStyles />
        </ThemeProvider>
      </Router>
    );
  }
}

export default hot(App);
