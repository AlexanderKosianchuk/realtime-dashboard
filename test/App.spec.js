import React from 'react';
import { App } from 'App';

function setup(ownProps = {}) {
  return shallow(<App {...ownProps} />, { attachTo: document.getElementById('react') });
}

describe('App', () => {
  const wrapper = setup();

  it('should render properly ', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
