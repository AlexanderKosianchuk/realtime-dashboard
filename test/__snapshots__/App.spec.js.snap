// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly  1`] = `
<Router history="{{...}}">
  <ThemeProvider theme="{{...}}">
    <Helmet defer="{false}"
            htmlattributes="{{...}}"
            encodespecialcharacters="{true}"
            defaulttitle="title"
            titletemplate="%s | title"
            titleattributes="{{...}}"
    >
    </Helmet>
    <Switch>
      <Route path="/"
             exact="{true}"
             component="{[Function:"
             home]}
      >
      </Route>
      <Route component="{[Function:"
             notfound]}
      >
      </Route>
    </Switch>
    <Connect(SystemAlerts)>
    </Connect(SystemAlerts)>
    <_default>
    </_default>
  </ThemeProvider>
</Router>
`;
